apps/client/src/components/Pages/dispositif/Edition/CustomNavbar/CustomNavbarEdit/PublishModal/CompleteContent/CompleteContent.tsx

```diff
@@ -35,43 +35,42 @@ const CompleteContent = (props: Props) => {
   const [keepTranslations, setKeepTranslations] = useState(false);
   const dispositif = useSelector(selectedDispositifSelector);
   const [textContent, setTextContent] = useState<Content[]>(
-    getTextContent(status, !!dispositif?.hasDraftVersion, undefined, user.admin),
+    getTextContent(status, !!dispositif?.hasDraftVersion, undefined),
   );
   const values = useWatch();
 
-  const [hasChanges, setHasChanges] = useState<boolean | null>(user.admin ? null : false); // check changes only for admins
+  const [hasChanges, setHasChanges] = useState<boolean | null>(null);
   const [{ loading }, getHasChanges] = useAsyncFn(() =>
-    dispositif?._id && user.admin
-      ? API.getDispositifHasTextChanges(dispositif?._id.toString())
-      : Promise.resolve(false),
+    dispositif?._id ? API.getDispositifHasTextChanges(dispositif?._id.toString()) : Promise.resolve(false),
   );
 
   const contentType = useContentType();
   const totalSteps = useMemo(() => getTotalSteps(contentType), [contentType]);
 
   // when form changes, reset hasChange
   useEffect(() => {
-    setHasChanges(user.admin ? null : false);
-  }, [values, user.admin]);
+    setHasChanges(null);
+  }, [values]);
 
   useEffect(() => {
     if (!loading && hasChanges === null) getHasChanges().then((res) => setHasChanges(res));
   }, [hasChanges, loading, getHasChanges]);
 
   useEffect(() => {
     if (hasChanges !== null) {
-      const textContent = getTextContent(status, !!dispositif?.hasDraftVersion, hasChanges, user.admin);
+      const textContent = getTextContent(status, !!dispositif?.hasDraftVersion, hasChanges);
```

Comment:
It compiles because `getTextContent` is called at the top with `undefined` explicitely, and here only if `hasChanges` is not null (so a boolean)

So in the function, it can only be a boolean or undefined. Do you think it's fine like this?